timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimeLockServiceMetadataIntegrationTest.java

```diff
@@ -0,0 +1,281 @@
+    private static final LockWatchStateUpdateVisitor LOCK_WATCH_STATE_UPDATE_VISITOR =
+            new LockWatchStateUpdateVisitor();
+
+    private static final String WATCHED_TABLE_NAME = "watched-table";
+    private static final LockDescriptor WATCHED_LOCK_1 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock1".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_2 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock2".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_3 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock3".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_4 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock4".getBytes(StandardCharsets.UTF_8));
+    private static final Map<LockDescriptor, ChangeMetadata> ALL_WATCHED_LOCKS_WITH_METADATA = ImmutableMap.of(
+            WATCHED_LOCK_1,
+            ChangeMetadata.unchanged(),
+            WATCHED_LOCK_2,
+            ChangeMetadata.updated("old".getBytes(StandardCharsets.UTF_8), "new".getBytes(StandardCharsets.UTF_8)),
+            WATCHED_LOCK_3,
+            ChangeMetadata.deleted("deleted".getBytes(StandardCharsets.UTF_8)),
+            WATCHED_LOCK_4,
+            ChangeMetadata.created("created".getBytes(StandardCharsets.UTF_8)));
+    private static final String UNWATCHED_TABLE_NAME = "a-random-unwatched-table";
+    private static final LockDescriptor UNWATCHED_LOCK_1 =
+            AtlasRowLockDescriptor.of(UNWATCHED_TABLE_NAME, "lock1".getBytes(StandardCharsets.UTF_8));
+    private static final IdentifiedLockRequest WATCHED_LOCK_REQUEST_WITH_METADATA =
+            standardRequestWithMetadata(ALL_WATCHED_LOCKS_WITH_METADATA);
+
+    private final LockLog lockLog = new LockLog(new MetricRegistry(), () -> 10000L);
+    private final ScheduledExecutorService scheduledExecutorService = PTExecutors.newSingleThreadScheduledExecutor();
+    private final AsyncLockService asyncLockService =
+            AsyncLockService.createDefault(lockLog, scheduledExecutorService, scheduledExecutorService);
+    private final AsyncTimelockServiceImpl timeLockService =
+            new AsyncTimelockServiceImpl(asyncLockService, new InMemoryTimestampService(), lockLog);
+    private final ConjureStartTransactionsRequest startTransactionsRequestWithInitialVersion =
+            ConjureStartTransactionsRequest.builder()
+                    .requestId(UUID.randomUUID())
+                    .lastKnownVersion(ConjureIdentifiedVersion.of(
+                            asyncLockService.leaderTime().id().id(), 0))
+                    .numTransactions(1)
+                    .requestorId(UUID.randomUUID())
+                    .build();
+
+    @Before
+    public void setupLockServiceWithWatches() {
+        timeLockService.startWatching(
```

Comment:
May be worth also testing an exact match scenario, rather than just prefix?

Reply:
Isn't this just testing what's already covered by `LockWatchingServiceImplTest`? I'd say we either duplicate all of the tests from there or none. thoughts?